src/main/java/ortus/boxlang/runtime/net/BoxHttpClient.java

```diff
@@ -1586,6 +1595,213 @@ private Object processBufferedResponse( HttpResponse<byte[]> response ) throws I
+
+						// Store error in httpResult
+						this.error				= true;
+						this.requestException	= e;
+						this.errorMessage		= e.getMessage();
+						this.httpResult.put( Key.errorDetail, "Streaming callback error: " + e.getMessage() );
+
+						// Call onError callback if provided
+						if ( onErrorCallback != null ) {
+							try {
+								context.invokeFunction( onErrorCallback, new Object[] { e, this.httpResult } );
+							} catch ( Exception callbackError ) {
+								streamLogger.error( "Error in onError callback during streaming", callbackError );
+							}
+						}
+						break; // Stop processing on error
+					}
+				}
+			}			// Finalize httpResult after streaming completes
+			String finalContent = accumulatedContent.toString();
+			this.httpResult.put( Key.fileContent, finalContent );
+			this.httpResult.put( Key.errorDetail, "" );
+			this.httpResult.put( Key.executionTime, Duration.between( this.startTime.toInstant(), Instant.now() ).toMillis() );
+
+			// Determine if response is text
+			boolean isText = false;
+			if ( contentType == null || contentType.isEmpty() ) {
+				isText = true;
+			} else {
+				String lowerContentType = contentType.toLowerCase();
+				isText = lowerContentType.startsWith( "text" )
+				    || lowerContentType.startsWith( "message" )
+				    || lowerContentType.equals( "application/octet-stream" );
+			}
+			this.httpResult.put( Key.text, isText );
 
+			// Handle output file saving if specified (streaming variant)
+			if ( this.outputDirectory != null ) {
+				if ( this.outputFile == null || this.outputFile.trim().isEmpty() ) {
+					// Extract filename from Content-Disposition or URL
+					String dispositionHeader = HttpResponseHelper.extractFirstHeaderByName( headers, Key.of( "content-disposition" ) );
+					if ( dispositionHeader != null ) {
+						Pattern	pattern	= Pattern.compile( "filename=\"?([^\";]+)\"?" );
+						Matcher	matcher	= pattern.matcher( dispositionHeader );
+						if ( matcher.find() ) {
+							this.outputFile = matcher.group( 1 );
+						}
+					}
+					if ( this.outputFile == null || this.outputFile.trim().isEmpty() ) {
+						this.outputFile = Path.of( this.targetHttpRequest.uri().getPath() ).getFileName().toString();
+					}
+					if ( this.outputFile == null || this.outputFile.trim().isEmpty() ) {
+						throw new BoxRuntimeException( "Unable to determine filename from response" );
+					}
+				}
+
+				String destinationPath = Path.of( this.outputDirectory, this.outputFile ).toAbsolutePath().toString();
+				FileSystemUtil.write( destinationPath, finalContent, charset, true );
+			}
+
+			// Announce the HTTP response now that streaming is complete
+			interceptorService.announce(
+			    BoxEvent.ON_HTTP_RESPONSE,
+			    ( java.util.function.Supplier<IStruct> ) () -> Struct.ofNonConcurrent(
+			        Key.requestID, this.requestID,
+			        Key.response, response,
+			        Key.result, this.httpResult,
+			        Key.of( "streaming" ), true,
```

Comment:
Shouldn't these streaming and chunkCount keys be added to the non-streaming `ON_HTTP_REQUEST` announcement?